core/src/Revolution/modResource.php

```diff
@@ -1584,4 +1584,38 @@ public function clearCache($context = '')
             $this->xpdo->invokeEvent('OnResourceCacheUpdate', ['id' => $this->get('id')]);
         }
     }
+
+    /**
+     * Returns array with all neighborhood resources
+     *
+     * @return array $arr Array with neighborhood from left and right
+     */
+    public function getNeighborhood()
+    {
+        $arr = [];
+
+        $q = $this->xpdo->newQuery(modResource::class, ['parent' => $this->parent]);
```

Comment:
I'm not sure I like this approach. On large sites, this basically means that everytime you open resource to edit you'll fetch all resources under the same parent, which could be hundreds or even thousands of records in some cases, just to get 2 specific IDs.

Wouldn't be better to query directly for the resource under and above the current one?

Reply:
It was idea for modx docs, not for modx CMS) my mistake

Reply:
Here's one way to get the prev/next without having to get all resources: add a `where` that matches the sortby.

For the next:

```
$qNext->where([
     'id:!=' => $this->get('id'),
     'AND:menuindex:<=' => $this->get('menuindex'),
]);
$qNext->limit(1);
$qNext->sortby('menuindex', 'ASC');
```

For the previous:
```
$qPrevious->where([
     'id:!=' => $this->get('id'),
     'AND:menuindex:>=' => $this->get('menuindex'),
]);
$qPrevious->limit(1);
$qPrevious->sortby('menuindex', 'ASC');
```

The "equals or larger/smaller" makes sure that items with the same menuindex still appear. 

It means two queries, but will be much more optimized than loading all siblings, especially on large sites.